src/Shared/CertificateGeneration/Paths.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.IO;
+using static System.Environment;
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Certificates.Generation
+{
+    internal static class Paths
+    {
+        public static string Home => Environment.GetFolderPath(SpecialFolder.MyDocuments);
+
+        public static string GetUserTempFile(string suffix = ".tmp")
+        {
+            string directory = Paths.XdgRuntimeDir ?? Home ?? // Should be user folders.
+                               Path.GetTempPath();            // Probably global on Linux.
+
+            return Path.Combine(directory, Guid.NewGuid() + suffix);
+        }
+
+        private static string? XdgRuntimeDir => Environment.GetEnvironmentVariable("XDG_RUNTIME_DIR");
```

Comment:
Why does https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-5.0 is not enough?

Reply:
It's not secure to use a word writable location like `/tmp`.
By the time you use the file, someone can replace it.

Reply:
I was wrong about this. `Path.GetTempFileName` is secure enough. It's created `rw` only for the current user, and only that user can remove the file.